Automattic/woocommerce-payments
# Add support for WooCommerce Deposits when using Apple Pay and Google Pay

Fixes #7907
Fixes woocommerce/woocommerce-deposits#569

#### Changes proposed in this Pull Request



This modifies the add to cart AJAX request made during express payments to include field names prefixed with `wc_`. This is to account for fields added by extensions, for example the deposits extensions fields `wc_deposit_option` and `wc_deposit_payment_plan`.

The goal is to include extension fields during an express payment request so if an input affects the price, that is reflected in the data added to the cart. 






#### Testing instructions

This has been tested with the deposits extension.

1. Enable Express Pay options (Google and Apple Pay) to be displayed on product pages.
1. Install and enable WooCommerce Deposits
3. Create a virtual simple product with an optional deposit of 50%
6. Visit virtual simple product with optional deposit of 50% 
   1. Select "Pay Deposit"
   2. Click Apple/Google Pay button
   3. The price in the platforms popover should reflect the deposit amount
   4. Proceed with purchase
   5. The deposit amount should be recorded as paid against the order
   6. The remaining amount should be recorded against the order
1. Update the Deposit options ensure it works as expected. Special attention to Enable Deposits and Default Deposit Selected Type.
    ![image](https://github.com/Automattic/woocommerce-payments/assets/12385501/98972762-8e97-4f1d-bb0d-e7011dd7ee0e)
7. Update the product to Physical (or create a new product) an check the price is not updated after the shipping methods load (more info about that in the second part of https://github.com/Automattic/woocommerce-payments/issues/7577#issuecomment-1847967971)
4. Create a virtual simple product with an optional payment plan 
8. Visit virtual simple product with optional payment plan
   1. Select "Pay Deposit"
   3. Select payment plan
   4. Click Apple/Google Pay button
   5. Proceed with purchase
   6. The first payment amount should be recorded as paid against the order
   7. The remaining amount should be recorded against the order








-------------------


**Post merge**
